### PR TITLE
containerlab: init at 0.41.2

### PR DESCRIPTION
**Pull request by Janik-Haag**
###### Description of changes

Closes #196282

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/containerlab/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  vendorHash = "sha256-dumXV5o/x4rqvX3tATsWFXz6DSowTSgaiHvzFoQaDIw=";
+
+  ldflags = [ "-s" "-w" "-X" "github.com/srl-labs/containerlab/cmd.version=${version}" "-X" "github.com/srl-labs/containerlab/cmd.commit=${src.rev}" "-X" "github.com/srl-labs/containerlab/cmd.date=1970-01-01T00:00:00Z" ];
+
+  meta = with lib; {
+    description = "container-based networking lab";
+    homepage = "https://containerlab.dev/";
+    changelog = "https://github.com/srl-labs/containerlab/releases/tag/v0.41.2";
```
Comment (natsukium) [inline pkgs/tools/networking/containerlab/default.nix:24]:
```suggestion
    changelog = "https://github.com/srl-labs/containerlab/releases/tag/v${version}";
```

File: pkgs/tools/networking/containerlab/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "srl-labs";
+    repo = "containerlab";
+    rev = "v${version}";
+    hash = "sha256-HOsfQUCYlU19C30XKCQZj2Mw5NQfmp9miZYmN1s14QU=";
+  };
+
+  vendorHash = "sha256-dumXV5o/x4rqvX3tATsWFXz6DSowTSgaiHvzFoQaDIw=";
+
+  ldflags = [ "-s" "-w" "-X" "github.com/srl-labs/containerlab/cmd.version=${version}" "-X" "github.com/srl-labs/containerlab/cmd.commit=${src.rev}" "-X" "github.com/srl-labs/containerlab/cmd.date=1970-01-01T00:00:00Z" ];
```
Comment (natsukium) [inline pkgs/tools/networking/containerlab/default.nix:19]:
```suggestion
  ldflags = [
    "-s"
    "-w"
    "-X github.com/srl-labs/containerlab/cmd.version=${version}"
    "-X github.com/srl-labs/containerlab/cmd.commit=${src.rev}"
    "-X github.com/srl-labs/containerlab/cmd.date=1970-01-01T00:00:00Z"
  ];
```
One argument per line is preferred.

File: pkgs/tools/networking/containerlab/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  meta = with lib; {
+    description = "container-based networking lab";
```
Comment (natsukium) [inline pkgs/tools/networking/containerlab/default.nix:22]:
```suggestion
    description = "Container-based networking lab";
```

File: pkgs/tools/networking/containerlab/default.nix (hunk 1 above)
Comment (emilylange) [reply pkgs/tools/networking/containerlab/default.nix:24]: That's within the `with lib;` scope.
It does work, but from what I can tell, the following is slightly more preferred (lately):

```suggestion
    changelog = "https://github.com/srl-labs/containerlab/releases/tag/${src.rev}";
```

Furthermore, you could also drop the `with lib;` and use

```nix
{
    license = lib.licenses.bsd3;
    maintainers = with lib.maintainers; [ janik ];
}
```

:)

Comment (Janik-Haag) [reply pkgs/tools/networking/containerlab/default.nix]: Is it fine now?

File: pkgs/tools/networking/containerlab/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    "-X" "github.com/srl-labs/containerlab/cmd.version=${version}"
+    "-X" "github.com/srl-labs/containerlab/cmd.commit=${src.rev}"
+    "-X" "github.com/srl-labs/containerlab/cmd.date=1970-01-01T00:00:00Z"
+  ];
+
+  meta = with lib; {
+    description = "Container-based networking lab";
+    homepage = "https://containerlab.dev/";
+    changelog = "https://github.com/srl-labs/containerlab/releases/tag/${src.rev}";
+    license = licenses.bsd3;
```
Comment (emilylange) [inline pkgs/tools/networking/containerlab/default.nix:31]: It looks like `darwin` is only supported by a workaround leveraging a docker container.
See https://containerlab.dev/install/#mac-os and https://github.com/srl-labs/containerlab/issues/577

This should fix the ofborg CI checks :)

```suggestion
    license = licenses.bsd3;
    platforms = platforms.linux;
```

**Review by emilylange — CHANGES_REQUESTED** [review]
Just one more small thing :innocent: 
But otherwise it looks good to me.
Thank you! :)

File: pkgs/tools/networking/containerlab/default.nix (hunk 4 above)
Comment (Janik-Haag) [reply pkgs/tools/networking/containerlab/default.nix:31]: fixed, I always forget about darwin :see_no_evil: 
Thank's for reviewing :heart:

**Review by emilylange — APPROVED** [review]

**Pull request merged**